internal/it/compact.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+func MakeCompactClusterConfig(clusterName, mapName, inMemoryFormat string, port int) string {
+	return fmt.Sprintf(`
+		<hazelcast xmlns="http://www.hazelcast.com/schema/config"
+			xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+			xsi:schemaLocation="http://www.hazelcast.com/schema/config
+			http://www.hazelcast.com/schema/config/hazelcast-config-4.0.xsd">
+			<cluster-name>%s</cluster-name>
+			<map name="%s">
+				<in-memory-format>%s</in-memory-format>
+			</map>
+			<network>
+			   <port>%d</port>
+			</network>
+		</hazelcast>
+	`, clusterName, mapName, inMemoryFormat, port)
+}
+
+type CompactTestContext struct {
+	T            *testing.T
+	M            *hazelcast.Map
+	Cluster      *TestCluster
+	Client1      *hazelcast.Client
+	Client2      *hazelcast.Client
+	MapName      string
+	InMemFmt     string
+	Serializers1 []serialization.CompactSerializer
+	Serializers2 []serialization.CompactSerializer
+}
+
+func (tcx CompactTestContext) Tester(f func(CompactTestContext)) {
+	ensureRemoteController(true)
+	runner := func(tcx CompactTestContext) {
+		if tcx.MapName == "" {
+			tcx.MapName = NewUniqueObjectName("map")
+		}
+		if tcx.Cluster == nil {
+			cn := fmt.Sprintf("%s-%s", tcx.T.Name(), tcx.InMemFmt)
+			port := NextPort()
+			cfg := MakeCompactClusterConfig(cn, tcx.MapName, tcx.InMemFmt, port)
+			tcx.Cluster = StartNewClusterWithConfig(2, cfg, port)
+		}
+		cfg1 := tcx.Cluster.DefaultConfig()
+		cfg1.Serialization.Compact.SetSerializers(tcx.Serializers1...)
+		tcx.T.Logf("map name: %s", tcx.MapName)
+		tcx.T.Logf("cluster address: %s", cfg1.Cluster.Network.Addresses[0])
+		if tcx.Client1 == nil {
+			tcx.Client1 = getDefaultClient(&cfg1)
+		}
+		cfg2 := tcx.Cluster.DefaultConfig()
+		cfg2.Serialization.Compact.SetSerializers(tcx.Serializers1...)
```

Comment:
should this be serializers2?

Reply:
Right, fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/910/commits/e6f38dc4f62fadf3654d60034195dc914ac20dc4